PalasoUIWindowsForms/ImageToolbox/ImageToolboxControl.cs

```diff
@@ -163,24 +158,27 @@ private ListViewItem AddControl(string label, Bitmap bitmap, string imageKey, Sy
 			return item;
 		}
 
-		bool _inIndexChanging	;
+		// Changed this to remember the selected index because something is causing the SelectedIndexChanged
+		// event to fire twice each time an icon is clicked.
+		int _previousSelectedIndex = -1;
 
 		private void listView1_SelectedIndexChanged(object sender, EventArgs e)
 		{
+			if (_toolListView.SelectedItems.Count == 0)
+			{
+				_previousSelectedIndex = -1;
+				return;
+			}
 
-			if (_inIndexChanging)
+			if (_previousSelectedIndex == _toolListView.SelectedIndices[0])
 				return;
 
 			try
 			{
-				_inIndexChanging = true;
-
-				if (_toolListView.SelectedItems.Count == 0)
```

Comment:
if `SelectedItems.Count == 0` is still a possibility (e.g. when there are no items available to select), an exception will be thrown outside the try block.